setup.py

```diff
@@ -41,6 +49,57 @@ def find_version(*file_paths):
     raise RuntimeError("Unable to find version string.")
 
 
+cmdclass = {}
+if SPHINX_AVAILABLE:
+
+    class MakeDocs(BuildDoc):
+        """Port of old `cylc-make-docs`. Removed in #2989.
+
+        This class extends the Sphinx command class for setuptools. With
+        a difference that it tries to mimic the behaviour of the previous
+        `cylc-make-docs`.
+
+        So first it will execute `make-commands.sh`, which builds the
+        commands help information, in the appendices.
+
+        Then, instead of calling one builder, this class will call the
+        builder fot he single HTML, and also the builder for the multiple
+        HTML documentation.
+
+        Finally, one more tweak in this class is to move the doctrees
+        folder (in the same level as the documentation) to within the
+        documentation folder, named `.doctrees`, as before with
+        `cylc-make-docs`.
+        """
+
+        def run(self):  # type: () -> None
+            try:
+                self.spawn(["./doc/src/custom/make-commands.sh"])
+            except DistutilsExecError as e:
+                self.warn("Failed to run make-commands.sh")
+                raise e
+            self.do_run("html", "built-sphinx")
```

Comment:
Hard-coding the builds was a temporary approach it would be good to get away from (Sphinx supports multiple builders out the box, there's no need for us to specify which ones users are permitted to use).

Does `cmdclass` let you define arguments i.e:

```python
def run(self, builder):
    # ...
    self.do_run(builder, builder)
```

Reply:
It does! We use a similar approach in `isodatetime` for PyTest. I believe you asked in another pull request about it, and I forgot to answer. You asked if we could pass parameters to pytest to enable/disable something I think?

It can be done via these `cmdclasses`, but it's a bit more obscure than simply adding the parameters to the function signature if I remember well.

>If it's anything other than a quick change we can do this later, it's more important to get this PR in!

If possible that would be great. Number of changes/commits here is getting hard to manage, especially after each rebase :disappointed_relieved:

Reply:
:+1: